components/arm/fake/fake.go

```diff
@@ -0,0 +1,186 @@
+// Package fake implements a fake arm.
+package fake
+
+import (
+	"context"
+	// for arm model.
+	_ "embed"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/components/arm"
+	"go.viam.com/rdk/components/arm/eva"
+	ur "go.viam.com/rdk/components/arm/universalrobots"
+	"go.viam.com/rdk/components/arm/xarm"
+	"go.viam.com/rdk/components/arm/yahboom"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+)
+
+// ModelName is the string used to refer to the fake arm model.
+const ModelName = "fake"
+
+//go:embed fake_model.json
+var fakeModelJSON []byte
+
+// AttrConfig is used for converting config attributes.
+type AttrConfig struct {
+	ArmModel string `json:"arm-model"`
+}
+
+func init() {
+	registry.RegisterComponent(arm.Subtype, ModelName, registry.Component{
+		Constructor: func(ctx context.Context, _ registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			if config.Attributes.Bool("fail_new", false) {
+				return nil, errors.New("whoops")
+			}
+			return NewArm(ctx, config, logger)
+		},
+	})
+
+	config.RegisterComponentAttributeMapConverter(arm.SubtypeName, ModelName,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf AttrConfig
+			return config.TransformAttributeMapToStruct(&conf, attributes)
+		},
+		&AttrConfig{},
+	)
+}
+
+// NewArm returns a new fake arm.
+func NewArm(ctx context.Context, cfg config.Component, logger golog.Logger) (arm.LocalArm, error) {
```

Comment:
I don't think we actually need `context` here, right?

Reply:
I see now we don't, this is an artifact from the old version that I didn't remove